Source/Extensions/ConstraintsSetup.swift

```diff
@@ -55,14 +66,92 @@ extension BottomContainerView {
     addConstraint(NSLayoutConstraint(item: topSeparator, attribute: .Height,
       relatedBy: .Equal, toItem: nil, attribute: .NotAnAttribute,
       multiplier: 1, constant: 1))
+
+  }
+
+  func setupCompactHeightConstraints() {
+    for attribute: NSLayoutAttribute in [.Height, .Left, .Top] {
+      addConstraint(NSLayoutConstraint(item: topSeparator, attribute: attribute,
+        relatedBy: .Equal, toItem: self, attribute: attribute,
+        multiplier: 1, constant: 0))
+    }
+
+    addConstraint(NSLayoutConstraint(item: doneButton, attribute: .CenterX,
```

Comment:
The problem I ran into with that approach is storage of references to the constraints. Since all these constraints are added in an extension, I don't have access to stored properties. I could put them in global state, but I figured adding and removing constraints this way was better than this option.

Another option would be to re-factor this code out of an extension and into the class implementations, allowing me to store references to the constraints in properties.

What do you think of those options? Do have a suggestion for a different option?